sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalog.scala

```diff
@@ -418,4 +419,10 @@ class JDBCTableCatalog extends TableCatalog
   private def toSQLId(ident: Identifier): String = {
     toSQLId(ident.namespace.toSeq :+ ident.name)
   }
+
+  protected def convertSchemaNameToPattern(schemaName: String): String = {
+    schemaName.replace("\\", "\\\\")
+        .replace("%", "\\%")
+        .replace("_", "\\_")
```

Comment:
are both only ones to escape? would be great if we add a comment for the spec

Reply:
Yes, those are defaults in SQL standard of almost all databases.
JDBC API does not specify exact format, https://docs.oracle.com/javase/8/docs/api/java/sql/DatabaseMetaData.html#getTables-java.lang.String-java.lang.String-java.lang.String-java.lang.String:A-

Responsibility of concrete JDBC driver implementation is to support patterns, so in theory, there maybe some JDBC drivers that use some different escape characters (the only one I know is Oracle, while Postgres, MySQL, teradata, Snowflake, Redshift, etc use only those two special characters).

Under the hood, JDBC drivers usually use LIKE operator for this. Like operators have only those two special characters.

Reply:
I also checked decompiled code of Microsoft SQL Server JDBC driver, they use brackets for escaping in LIKE operations, but actually, interface expects special characters escaped by \, so it seems every driver accepts this form.
But ofc, I made a method for conversion, so anyone can inherit JDBC Table Catalog and override this behaviour.